File: public/usage-examples/geometry/lines_from_triangle-1-example.cpp
#include "splashkit.h"
#include <vector>
#include <string>
#include <algorithm>

int main()
{
    open_window("Lines From Triangle", 400, 400);


    //Build the triangle and grab its 3 edges
    triangle tri    = triangle_from(100, 100, 200,  80, 150, 200);
    std::vector<line> lines = lines_from(tri);

    const float R = 10.0f;  //circle radius

    while (!window_close_requested("Lines From Triangle"))
    {
        process_events();
        clear_screen(COLOR_WHITE);

        //Mouse x and y position
        float mx = mouse_x();
        float my = mouse_y();

        //Draw each line, highlighting if touched
        for (size_t i = 0; i < lines.size(); ++i)
        {
            line ln = lines[i];

            float x1 = ln.start_point.x, y1 = ln.start_point.y;
            float x2 = ln.end_point.x,   y2 = ln.end_point.y;

            float left   = std::min(x1, x2) - R;
            float right  = std::max(x1, x2) + R;
            float top    = std::min(y1, y2) - R;
            float bottom = std::max(y1, y2) + R;

            bool overlap = mx >= left && mx <= right && my >= top  && my <= bottom;

            color col = overlap ? COLOR_BLUE : COLOR_RED;
            draw_line(col, ln);

            float midx = (x1 + x2) * 0.5f;
            float midy = (y1 + y2) * 0.5f;
            draw_text(std::to_string(i), COLOR_BLACK, midx, midy);
        }

        //Draw a little circle around the mouse
        draw_circle(COLOR_GREEN, mx, my, R);

        refresh_screen();
    }

    return 0;
}

File: public/usage-examples/geometry/lines_from_triangle-1-example.py
from splashkit import *

open_window("Lines From Triangle", 400, 400)

#Define the three corner points
p1 = point_at(100, 100)
p2 = point_at(200,  80)
p3 = point_at(150, 200)

#touch circle radius
R = 10

while not quit_requested():
    process_events()
    clear_screen(color_white())

    #Mouse coords
    mx, my = mouse_x(), mouse_y()

    #List of edges as point-pairs
    edges = [(p1, p2), (p2, p3), (p3, p1)]

    for idx, (a, b) in enumerate(edges):
        x1, y1 = a.x, a.y
        x2, y2 = b.x, b.y

        #A simple “touch zone” around the line plus radius
        left,  right  = min(x1, x2) - R, max(x1, x2) + R
        top,   bottom = min(y1, y2) - R, max(y1, y2) + R

        overlap      = (left <= mx <= right) and (top <= my <= bottom)

        #highlight colour blue if overlap or not in red
        col = color_blue() if overlap else color_red()
        draw_line(col, x1, y1, x2, y2)

        #Draw the index at the midpoint
        mid_x, mid_y = (x1 + x2) * 0.5, (y1 + y2) * 0.5
        draw_text_no_font_no_size(str(idx), color_black(), mid_x, mid_y)

    #Draw a little circle around the mouse
    draw_circle(color_green(), mx, my, R)

    refresh_screen()

close_all_windows()



